core/spec/models/spree/order_cancellations_spec.rb

```diff
@@ -1,6 +1,58 @@
 require 'spec_helper'
 
 describe Spree::OrderCancellations do
+  describe "#cancel_unit" do
+    subject { Spree::OrderCancellations.new(order).cancel_unit(inventory_unit) }
+    let(:order) { create(:shipped_order, line_items_count: 1) }
+    let(:inventory_unit) { order.inventory_units.first }
+
+    it "creates a UnitCancel record" do
+      expect { subject }.to change { Spree::UnitCancel.count }.by(1)
+      expect(subject.inventory_unit).to eq inventory_unit
+      expect(subject.reason).to eq Spree::UnitCancel::DEFAULT_REASON
+    end
+
+    it "cancels the inventory unit" do
+      expect { subject }.to change { inventory_unit.state }.to "canceled"
+    end
+
+    context "with a reason" do
```

Comment:
I see now that the expectation is listed above on line 12. I might prefer to to put these expectations in pairs:

```Ruby
context 'when a reason is specified' do
  it 'sets the reason to the reason specified`
end

context 'when a reason is not specified' do
  it 'leaves the reason as the default reason'
end
```